lib/source-code/source-code.js

```diff
@@ -411,19 +411,28 @@ class SourceCode extends TokenStore {
     }
 
     /**
-     * Determines if two tokens have at least one whitespace character
-     * between them. This completely disregards comments in making the
-     * determination, so comments count as zero-length substrings.
-     * @param {Token} first The token to check after.
-     * @param {Token} second The token to check before.
-     * @returns {boolean} True if there is only space between tokens, false
-     *  if there is anything other than whitespace between tokens.
+     * Determines if two nodes or tokens have at least one whitespace character
+     * between them.
+     * @param {ASTNode|Token} first The node or token to check after.
+     * @param {ASTNode|Token} second The node or token to check before.
+     * @returns {boolean} True if there is a whitespace character between
+     * any of the tokens found between the two given nodes or tokens.
      * @public
      */
     isSpaceBetweenTokens(first, second) {
-        const text = this.text.slice(first.range[1], second.range[0]);
+        let currentToken = this.getLastToken(first) || first;
 
-        return /\s/u.test(text.replace(/\/\*.*?\*\//gus, ""));
+        while (currentToken !== (this.getFirstToken(second) || second)) {
```

Comment:
Should we precompute the sentinel value before the loop, rather than recalculating that every loop iteration?